vasppy/kpoints.py

```diff
@@ -41,3 +41,51 @@ def __init__(
             self.shift = np.array([0.0, 0.0, 0.0])
         else:
             self.shift = shift
+
+def get_subdivisions_from_kspacing(kspacing: float, reciprocal_lattice_vectors: np.ndarray) -> tuple[int, int, int]:
+    """
+    Calculate subdivisions along reciprocal lattice vectors from the miniumum allowed distance between k-points (KSPACING).
+
+    Args:
+        kspacing (float): The minimum allowed distance between k-points.
+        reciprocal_lattice_vectors (np.ndarray): The reciprocal lattice vectors. These can be retrieved from ASE as atoms.cell.reciprocal() or from Pymatgen as
+        structure.lattice.reciprocal_lattice_crystallographic.matrix.
+
+    Returns:
+        tuple[int, int, int]: The subdivisions along each reciprocal lattice vector.
+    """
+    subdivisions = []
+    for row in reciprocal_lattice_vectors:
+        magnitude = np.linalg.norm(row)
+        subdivision = np.max([1, np.ceil((magnitude * 2 * np.pi) / kspacing)])
+
+        subdivisions.append(int(subdivision))
+
+    return tuple(subdivisions)
+
+def get_convergence_testing_kspacing(reciprocal_lattice_vectors: np.ndarray, kspacing_range: tuple[float, float]=(0.1, 0.8), step: float=0.02) -> tuple[float, ...]:
+    """
+    Generate a range of minimum allowed distances between k-points (KSPACING) for convergence testing. This function ensures that no two values of KSPACING
+    are generated that correspond to the same k-point mesh.
+
+    Args:
+        reciprocal_lattice_vectors (np.ndarray): The reciprocal lattice vectors. These can be retrieved from ASE as atoms.cell.reciprocal() or from Pymatgen as
+        structure.lattice.reciprocal_lattice_crystallographic.matrix.
+        kspacing_range (tuple[float, float]): The minimum and maximum KSPACING values. Defaults to (0.1, 0.8).
+        step (float): The interval between KSPACING values to be tested. Defaults to 0.02.
+
+    Returns:
+        list[float]: A range of KSPACING values which all correspond to distinct k-point grids.
+    """
+    allowed_kspacing = []
+    highest_total = 0
```

Comment:
```suggestion
    highest_total = 0.0
```
The current code initialises `highest_total` as an `int`, but then sets it as a `float` below (which `mypy` complains about).